types/src/collection.rs

```diff
@@ -1,37 +1,93 @@
-use sqlx::postgres::{PgHasArrayType, PgTypeInfo};
+use std::option::Option;
 use uuid::Uuid;
+use {
+    crate::async_graphql::{self, dataloader::DataLoader, Context, FieldResult},
+    crate::Database,
+};
 
-#[derive(Debug, Clone)]
-pub struct Collection {
-    pub title: String,
-
-    pub slug: String,
 
+/// Structure to represent an edited collection. Missing certain fields and chapters in it.
+/// Used for sending data to the front end
+#[derive(Debug, Clone, async_graphql::SimpleObject)]
+#[graphql(complex)]
+pub struct EditedCollection {
+    /// UUID for the collection
+    pub id: Uuid,
+    /// Full title of the collection
+    pub title: String,
+    /// ID of WordPress menu for navigating the collection
     pub wordpress_menu_id: Option<i64>,
+    /// URL slug for the collection, like "cwkw"
+    pub slug: String,
+}
 
-    pub chapters: Vec<Chapter>,
+/// Structure to represent a single chapter. Used to send data to the front end.
+#[derive(Debug, Clone, async_graphql::SimpleObject)]
+pub struct CollectionChapter {
+    /// UUID for the chapter
+    pub id: Uuid,
+    /// Full title of the chapter
+    pub title: String,
+    /// ID of WordPress page with text of the chapter
+    pub wordpress_id: std::option::Option<i64>,
+    /// Order within the parent chapter or collection
+    pub index_in_parent: i64,
+    /// Whether the chapter is an "Intro" or "Body" chapter
+    pub section: CollectionSection,
 }
 
-#[derive(Debug, Clone)]
-pub struct Chapter {
-    pub id: Option<Uuid>,
+/// Enum to represent the sections in an edited collection
+#[derive(async_graphql::Enum, Clone, Copy, PartialEq, Eq, Debug, sqlx::Type)]
+#[sqlx(type_name = "collection_section")]
+pub enum CollectionSection {
+    /// Intro chapter
+    Intro,
+    /// Body chapter
+    Body,
+}
 
-    pub url_slug: String,
+pub mod raw {
```

Comment:
Uh okay so thinking about this further, making a `pub mod raw` here is limiting because then other modules can't expose raw structs. So if we're gonna have a public `raw` module it should actually be a new folder with mirrored structure, so `types/src/collection.rs` and `types/src/raw/collection.rs` so that we could have other things inside that `raw` folder. Does that make sense to you?